x-pack/platform/plugins/shared/alerting/server/usage/lib/get_backfill_telemetry.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  AggregationsSingleMetricAggregateBase,
+  AggregationsTermsAggregateBase,
+  AggregationsStringTermsBucketKeys,
+} from '@elastic/elasticsearch/lib/api/types';
+import type { ElasticsearchClient, Logger } from '@kbn/core/server';
+import { getProviderAndActionFilterForTimeRange } from './get_telemetry_from_event_log';
+import { parseSimpleRuleTypeBucket } from './parse_simple_rule_type_bucket';
+import { parseAndLogError } from './parse_and_log_error';
+interface Opts {
+  esClient: ElasticsearchClient;
+  eventLogIndex: string;
+  logger: Logger;
+}
+
+interface GetBackfillTelemetryPerDayCountResults {
+  hasErrors: boolean;
+  errorMessage?: string;
+  countExecutions: number;
+  countBackfillsByExecutionStatus: Record<string, number>;
+  countGaps: number;
+  totalUnfilledGapDurationMs: number;
+  totalFilledGapDurationMs: number;
+}
+
+interface GetBackfillExecutionsPerDayCountResults {
+  hasErrors: boolean;
+  errorMessage?: string;
+  countExecutions: number;
+  countBackfillsByExecutionStatus: Record<string, number>;
+}
+
+interface GetGapDataPerDayCountResults {
+  hasErrors: boolean;
+  errorMessage?: string;
+  countGaps: number;
+  totalUnfilledGapDurationMs: number;
+  totalFilledGapDurationMs: number;
+}
+
+async function getBackfillExecutionCount({
+  esClient,
+  eventLogIndex,
+  logger,
+}: Opts): Promise<GetBackfillExecutionsPerDayCountResults> {
+  try {
+    const query = {
+      index: eventLogIndex,
+      size: 0,
+      query: getProviderAndActionFilterForTimeRange('execute-backfill'),
+      aggs: {
+        by_execution_status: {
+          terms: {
+            field: 'event.outcome',
+          },
+        },
+      },
+    };
+
+    const results = await esClient.search(query);
+
+    const totalBackfillExecutions =
+      typeof results.hits.total === 'number' ? results.hits.total : results.hits.total?.value;
+
+    const aggregations = results.aggregations as {
+      by_execution_status: AggregationsTermsAggregateBase<AggregationsStringTermsBucketKeys>;
+    };
+
+    return {
+      hasErrors: false,
+      countExecutions: totalBackfillExecutions ?? 0,
+      countBackfillsByExecutionStatus: parseSimpleRuleTypeBucket(
+        aggregations.by_execution_status.buckets
+      ),
+    };
+  } catch (err) {
+    const errorMessage = parseAndLogError(err, `getBackfillExecutionCount`, logger);
+
+    return {
+      hasErrors: true,
+      errorMessage,
+      countExecutions: 0,
+      countBackfillsByExecutionStatus: {},
+    };
+  }
+}
+
+async function getGapInfo({
+  esClient,
+  eventLogIndex,
+  logger,
+}: Opts): Promise<GetGapDataPerDayCountResults> {
+  try {
+    const query = {
+      index: eventLogIndex,
```

Comment:
We probably need to use `track_total_hits: true` because it will now return hits.total as a maximum of 10,000.

Reply:
Good catch! Updated in https://github.com/elastic/kibana/pull/221847/commits/1a9ac57bc2a68c22987532663442026bfe5538bd